force-app/main/default/classes/SearchContactsonAccountExtension.cls

```diff
@@ -0,0 +1,78 @@
+public with sharing class SearchContactsonAccountExtension {
+
+    public string selectedContactId {get;set;} 
+
+	public List<Contact> accountContacts{
+        get{
+            if(childContacts == null){
+                childContacts = [SELECT
+                                 Name,
+                                 Title,
+                                 Phone,
+                                 Is_Primary_Contact__c 
+                                 FROM Contact 
+                                 WHERE AccountId = :accountRecord.id
+                                 LIMIT 1000];
+            }
+            return childContacts;            
+        }
+        set;
+    }    
+
+    public string searchText
+    {
+        get{
+            if (searchText==null){
+                searchText = '';
+            }
+            return searchText;
+        }
+        set;
+    }
+
+    private List<Contact> childContacts;
+    private Account accountRecord {
+        get{
+            if(accountRecord == null){
+                accountRecord = (Account) sdtController.getRecord();
+            }
+            return accountRecord;
+        }
+        set;
+    }
+    private ApexPages.StandardController sdtController;
+
+    public SearchContactsonAccountExtension(ApexPages.standardController sdtController) {
+
+        this.sdtController = sdtController;
+        accountRecord = (Account) sdtController.getRecord();
```

Comment:
Please remove this line, since each time you will access accountRecord, the getter will be called.